Samples/BackgroundBlur/BackgroundBlur/Helpers/CSampleQueue.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <windows.h>
+#include <Mfidl.h>
+#include "common.h"
+#include <mutex> 
+
+using namespace MediaFoundationSamples;
+
+class CSampleQueue: public IUnknown
```

Comment:
>CSampleQueue [](http://example.com/codeflow?start=6&length=12)

Why not just use stl::list?

Reply:
Mainly because CSampleQueue came from demo code and because it's thread-safe without having to add lock guards around access to stl::list.